subsys/bluetooth/controller/hci/hci.c

```diff
@@ -1824,12 +1824,58 @@ static void le_conn_complete(struct pdu_data *pdu_data, u16_t handle,
 	struct radio_le_conn_cmplt *radio_cc;
 
 	if (!(event_mask & BT_EVT_MASK_LE_META_EVENT) ||
-	    !(le_event_mask & BT_EVT_MASK_LE_CONN_COMPLETE)) {
+	    (!(le_event_mask & BT_EVT_MASK_LE_CONN_COMPLETE) &&
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_PRIVACY)
+	     !(le_event_mask & BT_EVT_MASK_LE_ENH_CONN_COMPLETE))) {
+#else
+	     1)) {
+#endif /* CONFIG_BLUETOOTH_CONTROLLER_PRIVACY */
 		return;
 	}
 
 	radio_cc = (struct radio_le_conn_cmplt *) (pdu_data->payload.lldata);
 
+	if (!radio_cc->status) {
+		conn_count++;
+	}
+
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_PRIVACY)
+	if (le_event_mask & BT_EVT_MASK_LE_ENH_CONN_COMPLETE) {
+		struct bt_hci_evt_le_enh_conn_complete *sepe;
+
+		sepe = meta_evt(buf, BT_HCI_EVT_LE_ENH_CONN_COMPLETE,
+				sizeof(*sepe));
+
+		sepe->status = radio_cc->status;
+		sepe->handle = sys_cpu_to_le16(handle);
+		sepe->role = radio_cc->role;
+
+		sepe->peer_addr.type = radio_cc->peer_addr_type;
+		memcpy(&sepe->peer_addr.a.val[0], &radio_cc->peer_addr[0],
+		       BDADDR_SIZE);
+
+		/* Note: this could be an RPA set as the random address by
+		 * the Host instead of generated by the controller. That said,
+		 * this should make no difference. */
+		if ((radio_cc->own_addr_type) &&
+		    ((radio_cc->own_addr[5] & 0xc0) == 0x40)) {
+			memcpy(&sepe->local_rpa.val[0], &radio_cc->own_addr[0],
+			       BDADDR_SIZE);
+		} else {
+			memset(&sepe->local_rpa.val[0], 0x0, BDADDR_SIZE);
+		}
+
+		memcpy(&sepe->peer_rpa.val[0], &radio_cc->peer_rpa[0],
+		       BDADDR_SIZE);
+
+		sep->interval = sys_cpu_to_le16(radio_cc->interval);
+		sep->latency = sys_cpu_to_le16(radio_cc->latency);
+		sep->supv_timeout = sys_cpu_to_le16(radio_cc->timeout);
+		sep->clock_accuracy = radio_cc->mca;
```

Comment:
Yep, should be sepe. It gets sent because it's an offset pointer to the buf that you pass as a parameter to meta_evt().

I agree that sep is not a great name, it was supposed to be "sub-event pointer".